p2p/peering.go

```diff
@@ -1678,3 +1724,43 @@ func (rp *RemotePeer) SendHeadersSent() bool {
 	rp.requestedHeadersMu.Unlock()
 	return sent
 }
+
+// GetInitState attempts to get initial state by sending a GetInitState message.
+func (rp *RemotePeer) GetInitState(ctx context.Context) (*wire.MsgInitState, error) {
+	const opf = "remotepeer(%v).GetInitState"
+	msg := wire.NewMsgGetInitState()
+	msg.AddTypes(wire.InitStateTSpends)
+
+	c := make(chan *wire.MsgInitState, 1)
+	newRequest := rp.addRequestedInitState(c)
+	if !newRequest {
+		op := errors.Opf(opf, rp.raddr)
+		return nil, errors.E(op, errors.Invalid, "init state is already being requested from this peer")
+	}
+
+	stalled := time.NewTimer(stallTimeout)
+	out := rp.out
+	for {
+		select {
+		case <-ctx.Done():
+			go func() {
+				<-stalled.C
+				rp.deleteRequestedInitState()
+			}()
+			return nil, ctx.Err()
+		case <-stalled.C:
+			op := errors.Opf(opf, rp.raddr)
+			err := errors.E(op, errors.IO, "peer appears stalled")
+			rp.Disconnect(err)
+			return nil, err
+		case <-rp.errc:
+			stalled.Stop()
```

Comment:
This potentially leaks timer resources forever ([check the doc for the Stop() function](https://pkg.go.dev/time#Timer.Stop)).

You need to call this like the doc explains:

```go
if !stalled.Stop() {
	<-stalled.C
}
```